### PR TITLE
[StructuralMechanicsApplication] Adding CalculateCharacteristicLengthOnReferenceConfiguration method to CL utilities

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
Adding CalculateCharacteristicLengthOnReferenceConfiguration method to CL utilities

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp [hunk 1]
```diff
@@ -316,6 +316,30 @@ double ConstitutiveLawUtilities<TVoigtSize>::CalculateCharacteristicLength(const
...
 
+template<SizeType TVoigtSize>
+double ConstitutiveLawUtilities<TVoigtSize>::CalculateCharacteristicLengthOnReferenceConfiguration(const GeometryType& rGeometry)
+{
+    double radius = 0.0;
+
+    const SizeType points_number = rGeometry.size();
+	array_1d<double, 3> center = rGeometry[0].GetInitialPosition().Coordinates();
+    for ( IndexType i_node = 1 ; i_node < points_number ; ++i_node ) {
+        center += rGeometry[i_node].GetInitialPosition().Coordinates();
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp:327]: why don't you use the `Center`-Method of the geometry?

**Review by philbucher — CHANGES_REQUESTED** [review]

File: applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp (hunk 1 above)
Comment (philbucher) [reply applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp:327]: is it because you use the ref-configuration?

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp:327]: Yes, it is

**Review by loumalouomega — DISMISSED** [review]
Ok for me

**Review by philbucher — DISMISSED** [review]
ok I get it, sorry for the noise

Could you add a test please?

**Comment by loumalouomega** [comment]
> ok I get it, sorry for the noise
> 
> Could you add a test please?

On it

**Comment by loumalouomega** [comment]
Done @philbucher

**Review by loumalouomega — DISMISSED** [review]

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_constitutive_law_utilities.cpp [hunk 2]
```diff
@@ -187,9 +195,40 @@ KRATOS_TEST_CASE_IN_SUITE(ConstitutiveLawUtilitiesPolarDecomposition, KratosStru
...
+
+    // Compute CalculateCharacteristicLength
+    const double length = ConstitutiveLawUtilities<3>::CalculateCharacteristicLength(triangle);
+
+    KRATOS_CHECK_NEAR(length,  0.745356, tolerance);
+
+    // Compute CalculateCharacteristicLengthOnReferenceConfiguration
+    const double length_origin = ConstitutiveLawUtilities<3>::CalculateCharacteristicLengthOnReferenceConfiguration(triangle);
+    
+    KRATOS_CHECK_NEAR(length_origin,  0.745356, tolerance);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_constitutive_law_utilities.cpp:235]: could you set an updated configuration such that this method returns a different value?

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_constitutive_law_utilities.cpp:235]: Ok

**Comment by loumalouomega** [comment]
Done

**Review by loumalouomega — APPROVED** [review]

**Review by philbucher — APPROVED** [review]
nice :+1:

**Pull request merged**